RIOT-OS/RIOT
# makefiles/tools: Allow make reset via avrdude

### Contribution description

This PR adds the `make reset` capability for all ATmega based boards

#### Implementation Details

In order to flash AVR devices, avrdude needs the ability to reset them: Those using a bootloader will only enter it after a reset, and those programmed via ISP also need a reset to enter ISP mode.

Sadly, avrdude has no option to reset the board. But running it without commands will read and print the boards identification and fuse settings. For this, as reset is needed. This commit uses this side-effect reset to implement make reset for all AVR based boards


### Testing procedure

Run `make BOARD=<SOME_ATMEGA_BOARD> reset` and verify that the board reboots. This should work with both ISP programmed boards and those with bootloaders (e.g. Arduinos). Please keep in mind that only one process at a time should have access to the serial; so don't have `make term` running for devices using a bootloader to verify the reset.

### Issues/PRs references

None
